internal/inventory/model_provenance.go

```
@@ -0,0 +1,762 @@
+		root = canonicalOwner + "/" + name
+	}
+	return boundedProvenanceField(root, maxModelRootBytes)
+}
+
+func cleanDerivedModelName(value string) string {
+	value = strings.TrimSpace(value)
+	lower := strings.ToLower(value)
+	for _, ext := range []string{".safetensors", ".gguf", ".ggml", ".onnx", ".ort", ".bin"} {
+		if strings.HasSuffix(lower, ext) {
+			value = value[:len(value)-len(ext)]
+			lower = strings.ToLower(value)
```

golangci-lint (2.12.2)

[error] ineffectual assignment to lower

(ineffassign)